src/amo/components/App/index.js

```diff
@@ -49,8 +45,17 @@ interface MozNavigator extends Navigator {
   mozAddonManager?: Object;
 }
 
+// This function assumed the home page will no more than two parts in its path,
+// whereas any page other than the home page will have more than two.
+// Every path in the app starts with /lang/application/ (e.g., /en-US/firefox/)
+// which is what this logic is based upon.
+export const isHomePage = (location: ReactRouterLocationType) => {
+  return location.pathname.split('/').filter(Boolean).length <= 2;
```

Comment:
It still seems pretty fragile to me. I don't think it will work for the [`about`](https://github.com/mozilla/addons-frontend/blob/70774cf72261d6651d83bf6aad08bcda600cf6e5/src/amo/components/Routes/index.js#L70) or [`review_guide`](https://github.com/mozilla/addons-frontend/blob/70774cf72261d6651d83bf6aad08bcda600cf6e5/src/amo/components/Routes/index.js#L72) pages since those also have two paths.

Also, farfetched, but what if someone adds `basename` (a path prefix) to [Router](https://reacttraining.com/react-router/web/api/BrowserRouter), which might be necessary for development of some kind? 

There are just a lot of moving parts here. 

I still think moving some logic from `<App>` into `<Page>` would be the best way to go.

Reply:
Fair enough. What if I file another follow-up to fix this, but we land this in order to both remove the `viewContext` reducer hack and also to address any glitches with the loading skeletons? This would enable us to land all of the features needed for the new home page (which apparently is desired to be available in prod by the 22nd), and then I can take some time to address the follow-up.

Reply:
The About and Review Guide pages look pretty broken (too wide, missing top margin) so I'd rather not land it unless we have to. We could also revert the reducer patch to buy some time.

<details>
<summary>Screenshots</summary>

<img width="1833" alt="Screenshot 2019-10-04 11 50 41" src="https://user-images.githubusercontent.com/55398/66225187-70e7bc80-e69d-11e9-917f-16d0007103cf.png">
<img width="1877" alt="Screenshot 2019-10-04 11 50 52" src="https://user-images.githubusercontent.com/55398/66225188-70e7bc80-e69d-11e9-8f09-730d1efe0f93.png">


</details>